src/list/list.c

```diff
@@ -271,6 +277,8 @@ void list_unlink(struct le *le)
 	le->next = NULL;
 	le->prev = NULL;
 	le->list = NULL;
+
+	--list->cnt;
```

Comment:
perhaps add a check that cnt does not wrap below zero, in case the counter
accidentally gets out of balance ...